### PR TITLE
Create config file deprecation mechanism & rename Details->Preview

### LINKED ISSUE #2327: Inconsistent naming in settings: "preview panel"/"details view"
## Expected Behavior
The entry overview, beneath a database's entries in the main view should be named consistently throughout the application.

## Current Behavior
In `Tools->Settings->General->Entry Management`, this component is referred to as the "Details view".
In `Tools->Settings->General->Security->Convenience` this component is referred to as the "preview panel".  

## Possible Solution
One convention is chosen and used in both contexts, as well as any others I may have missed. I feel as though "entry details panel" and "entry preview panel" are both clear.

## Context
When I saw the "Hide the Details view" option in the Application Settings window, I was unsure what it did until I checked it, applied the settings, and left the settings menu.  This is mostly because I knew  the component in question as the "preview panel" from the "Hide passwords in the preview panel" option

## Debug Info

KeePassXC - Version 2.3.4
Revision: 6fe821c

Libraries:
- Qt 5.11.2
- libgcrypt 1.8.3

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 4.18.9-arch1-1-ARCH

Enabled extensions:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

(note: present in `develop` branch as well)

### PR DESCRIPTION
**Pull request by kneitinger**
## Description

- Changed `Settings->General->Entry Management->Hide the Details view` to `Hide the entry preview panel`
- Changed `Settings->Security->Convenience->Hide passwords in the preview panel` to `Hide passwords in the entry preview panel`
- Renamed widgets/variables associated with settings above from using `detail(s)` to `preview`
- Renamed `DetailsWidget` to `EntryPreviewWidget`
- Added upgrade/removal mechanism for config file entries:
  - Created new file mapping of deprecated setting names to new setting names
  - Upon loading settings in `Config::init()`, a new function `Config::upgrade()` is called that replaces the old setting with the new name (using the existing value)
  - If the new name is set to empty string, the setting is removed from the config
  - Entries added to the deprecation map for all of the settings I changed for this PR, as well as the one I changed in #2309.

## Motivation and context

Fixes #2327 
Enables future clean up and modification of config file names

## How has this been tested?

- Verified strings are updated in UI:
  
  
- Added entry to config deprecation map `{"security/IconDownloadFallbackToDuckDuckGo", ""},` to test removal of a nonexistent setting that was created in the ini file during development of a different issue.  Recompiled.
- Copied old `keepassxc.ini` to `/tmp`, and started application and closed it.
```
$ cp ../kneitnotes/keepassxc.ini /tmp/pre_test_keepassxc.ini
$ ./src/keepassxc --config ../kneitnotes/keepassxc.ini ../kneitnotes/testdb.kdbx
$ diff /tmp/pre_test_keepassxc.ini ../kneitnotes/keepassxc.ini 
44d46
< DetailSplitterState=442, 188
47d48
< HideDetailsView=false
48a50
> HidePreviewPanel=false
56a59
> PreviewSplitterState=442, 188
97,98c100,101
< IconDownloadFallbackToDuckDuckGo=true
< IconDownloadFallbackToGoogle=true
---
> HidePasswordPreviewPanel=true
> IconDownloadFallback=true
103d105
< hidepassworddetails=true
```
- Removed `{"security/IconDownloadFallbackToDuckDuckGo", ""},` map entry and recompiled.
- Restored old config, started application, and verified that the renamed details/preview settings have intended functionality still.

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
This is great work!

File: src/core/Config.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ class Config : public QObject
     void set(const QString& key, const QVariant& value);
     bool hasAccessError();
     void sync();
+    void upgrade();
```
Comment (droidmonkey) [inline src/core/Config.h]: This should be a private function

File: src/core/ConfigDeprecations.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#ifndef KEEPASSX_CONFIGDEPRECATIONS_H
+#define KEEPASSX_CONFIGDEPRECATIONS_H
+
+/*
+ * Map of configuration file settings that are either deprecated, or have
+ * had their name changed.  Entries in the map are of the form
+ *     {oldName, newName}
+ * Set newName to "REMOVED" to remove the setting from the file.
+ */
+static const QMap<QString, QString> deprecationMap = {
```
Comment (droidmonkey) [inline src/core/ConfigDeprecations.h]: Just move this directly into the Config.cpp file at the top. There is no need for this to be in a separate header.

File: src/core/Config.cpp [hunk 3]
```diff
@@ -63,6 +64,19 @@ void Config::sync()
     m_settings->sync();
 }
 
+void Config::upgrade()
+{
+    for (auto setting : deprecationMap.keys()) {
```
Comment (droidmonkey) [inline src/core/Config.cpp]: const auto& setting

File: src/core/Config.cpp [hunk 4]
```diff
@@ -63,6 +64,19 @@ void Config::sync()
...
+        if (m_settings->contains(setting)) {
+            if(deprecationMap.value(setting) != "REMOVED") {
```
Comment (droidmonkey) [inline src/core/Config.cpp]: I do not like "magic values". I would prefer it to just test for empty (ie. !deprecationMap.value(setting).isEmpty()) which means it was deleted.

**Comment by kneitinger** [comment]
Thanks for the feedback! Addressed it,  retested, and updated PR write up :+1:

**Comment by TheZ3ro** [comment]
Sorry if I'm late but I really like the deprecationMap and the upgrade function, really nice

**Comment by droidmonkey** [comment]
The party is over, but you can have some leftover cake 🍰

**Pull request merged**